### PR TITLE
TSLint retrofit for spellchecking and move to package files

### PR DESCRIPTION
**Pull request by szul**
Moved the TSLint from global package implementation to individual package implementation until we have the whole library converted appropriately. Used @cloudguy-pro's spellchecker as a test bed, since it has the smallest code base. Will convert the others as we progress. All new packages should use the TSLint.

@iMicknl based the TSLint on the same one that the Microsoft Bot Framework team uses, but various rules can be up for discussion.

### CODE REVIEW

**Comment by szul** [comment]
@cloudguy-pro Just check to make sure I didn't mess your code up by implementing the TSLint rules. Biggest thing was that these rules don't allow for `console.log()`. That can be up for discussion.

File: libraries/botbuilder-spell-check-middleware/package.json [hunk 1]
```diff
@@ -37,6 +37,12 @@
     "azure-cognitiveservices-spellcheck": "^1.0.1",
     "botbuilder": "^4.1.3",
     "restify": "^7.2.2",
+    "typescript": "^3.2.2",
```
Comment (iMicknl) [inline libraries/botbuilder-spell-check-middleware/package.json]: Should Typescript not be a dev-dependency?

Comment (szul) [reply libraries/botbuilder-spell-check-middleware/package.json]: Changed.

**Comment by cloudguy-pro** [comment]
Should be fine @szul looks good!

**Pull request merged**